src/main/java/net/mcreator/ui/minecraft/states/item/JItemPropertiesStatesList.java

```diff
@@ -202,17 +202,22 @@ void removeProperty(JItemPropertiesListEntry entry) {
 		propertyEntries.repaint();
 
 		PropertyData.NumberType data = entry.toPropertyData();
+
 		Set<StateMap> duplicateFilter = new HashSet<>();
-		statesList.stream().toList().forEach(s -> {

```

Comment:
Wasn't `<...>.stream().toList().<...>` enough to avoid concurrent modifications?

Reply:
According to some sources, the original link is maintained in streams in some cases so it is better to use iterator which is by spec recommended way for this